lib/oli/publishing/delivery_resolver.ex

```diff
@@ -279,4 +279,63 @@ defmodule Oli.Publishing.DeliveryResolver do
       select: spp.publication_id
     )
   end
+
+  @doc """
+  Returns the current revisions of all page resources whose purpose type matches the one it receives as parameter
+  ## Examples
+      iex> get_by_purpose(valid_section_slug, valid_purpose)
+      [%Revision{}, ...]
+
+      iex> get_by_purpose(invalid_section_slug, invalid_purpose)
+      []
+  """
+
+  def get_by_purpose(section_slug, purpose) do
+    Repo.all(
+      from(
+        revision in Revision,
+        join: sect_res in SectionResource,
+        on: sect_res.resource_id == revision.resource_id,
+        join: sec in Section,
+        on: sec.id == sect_res.section_id,
```

Comment:
This implementation isn't correct. It is going to return *all* revisions for each resource in a course.  What we need instead is it to return the exact published revision for each resource in a course. 

Take a look at how the other functions in this resolver leverage `section_resource_revisions` fragment builder to query through the `SectionsProjectsPublications` and `PublishedResource` schemas.  You can almost certainly delegate to that function in both of these two new resolver functions.

Again, we don't want to hardcode the resource type.

Reply:
Excellent. I will pay attention to what you tell me!